sympy/core/tests/test_expr.py

```diff
@@ -1011,7 +1011,6 @@ def test_extractions():
     assert (Rational(1, 2)*x).extract_multiplicatively(3) == x/6
     assert (sqrt(x)).extract_multiplicatively(x) is None
     assert (sqrt(x)).extract_multiplicatively(1/x) is None
-    assert x.extract_multiplicatively(-x) is None
```

Comment:
`x.extract_multiplicatively(c)` where `c` is `-1` or `-x` both fail because "the properties of `self` are not preserved" (as suggested in the docstring). I would recommend not spending more time on this as the other PR looks simpler and more correct at this point.